File: app/src/main/java/com/lingo/router/MainActivity.java
package com.lingo.router;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.lingo.grade.R;
import com.lingo.router.annotations.Destination;

@Destination(url = "router://page/home2", description = "Java首页")
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
    }
}

File: app/src/main/java/com/lingo/grade/MainActivity.kt
package com.lingo.grade

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.lingo.router.annotations.Destination
import com.lingo.router.runtime.Router

@Destination(url = "router://page/home", description = "首页")
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(view: View) {
        Router.go(this, "router://page/profile?name=lingo&msg=hehe")
    }
}